velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -62,57 +67,355 @@ void generateJsonTyped(
+    castToJsonFromRow(input, context, rows, flatResult);
+  } else {
+    VELOX_NYI(
+        "Casting from {} to JSON is not supported yet.",
+        input.type()->toString());
+  }
+}
+
+// Calculate the sum of Json string sizes of all elements within the range
+// [offset, offset + size) in vector wrapped with indices.
+FOLLY_ALWAYS_INLINE size_t getSumOfElementsStringSize(
+    const FlatVector<StringView>& vector,
+    const vector_size_t* indices,
+    vector_size_t offset,
+    vector_size_t size) {
+  size_t sum = 0;
+  for (auto i = offset; i < offset + size; ++i) {
+    sum += vector.valueAt(indices[i]).size();
+  }
+
+  return sum;
+}
+
+void castToJsonFromArray(
+    const BaseVector& input,
+    exec::EvalCtx* context,
+    const SelectivityVector& rows,
+    FlatVector<StringView>& flatResult) {
+  // input is guaranteed to be in flat encoding when passed in.
+  auto inputArray = input.as<ArrayVector>();
+
+  auto elements = inputArray->elements();
+  auto elementsRows =
+      functions::toElementRows(elements->size(), rows, inputArray);
+  exec::LocalDecodedVector elementsHolder(context, *elements, elementsRows);
+  auto elementsDecoded = elementsHolder.get();
+
+  exec::LocalSelectivityVector elementsBaseRows(
+      context->execCtx(), elementsDecoded->base()->size());
+  elementsBaseRows->clearAll();
+  elementsRows.applyToSelected([&](auto row) {
+    elementsBaseRows->setValid(elementsDecoded->index(row), true);
+  });
+  elementsBaseRows->updateBounds();
+
+  VectorPtr elementsResult;
+  BaseVector::ensureWritable(
+      *elementsBaseRows, JSON(), context->pool(), &elementsResult);
+  auto elementsStringResult = elementsResult->as<FlatVector<StringView>>();
+
+  VELOX_DYNAMIC_TYPE_DISPATCH(
+      castToJson,
+      elements->typeKind(),
+      *elementsDecoded->base(),
+      context,
+      *elementsBaseRows,
+      *elementsStringResult);
+
+  // Construct the Json string of each array from Json strings of its elements.
+  rows.applyToSelected([&](auto row) {
+    auto offset = inputArray->offsetAt(row);
+    auto size = inputArray->sizeAt(row);
+
+    if (size == 0) {
+      flatResult.set(row, "[]");
+    } else {
+      auto elementsStringSize = getSumOfElementsStringSize(
+          *elementsStringResult, elementsDecoded->indices(), offset, size);
+
+      auto proxy = exec::StringProxy<FlatVector<StringView>>(&flatResult, row);
+
+      // Reserves space for Json strings of all elements of this array, commas,
+      // and the enclosing square brackets.
+      proxy.reserve(elementsStringSize + size + 1);
+
+      proxy.append("["_sv);
+      proxy.append(
+          elementsStringResult->valueAt(elementsDecoded->index(offset)));
+      for (int i = offset + 1; i < offset + size; ++i) {
+        proxy.append(","_sv);
+        proxy.append(elementsStringResult->valueAt(elementsDecoded->index(i)));
+      }
+      proxy.append("]"_sv);
+
+      proxy.finalize();
+    }
+  });
+}
+
+void castToJsonFromMap(
```

Comment:
This function is ~100 lines -- not too bad, but probably could be shorter.

Reply:
I tried to reduce the number of statements by extracting some repeated code into helper functions. But then linter formatted the code in such a way that makes the code looks longer...